anthos-bm-apigee/docs/quickstart.md

```diff
@@ -0,0 +1,590 @@
+gcloud auth activate-service-account --key-file anthos-bm-owner.json
+```
+
+2. Clone this repo into the workstation from where the rest of this guide will be followed
+3. Update the `terraform.tfvars.sample` file to include variables specific to your environment
+```
+project_id                    = "<GOOGLE_CLOUD_PROJECT_ID>"
+region                        = "<GOOGLE_CLOUD_REGION_TO_USE>"
+zone                          = "<GOOGLE_CLOUD_ZONE_TO_USE>"
+credentials_file              = "<PATH_TO_GOOGLE_CLOUD_SERVICE_ACCOUNT_FILE>"
+#gcp_login_accounts           = ["<GCP_ACCOUNT_1>", <GCP_ACCOUNT_2>, <GCP_ACCOUNT_3>]
+username                      = tfadmin
+#gce_vm_service_account        = "<GCP VM Service Account>"
+```
+Uncomment the gce_vm_service_account if you are using custom compute engine service account
+
+An example of these configuration looks like this below:
+
+```
+project_id                     = "anthos-bm-example1"
+region                         = "us-central1"
+zone                           = "us-central1-a"
+credentials_file               = "anthos-bm-owner.json"
+username                      = tfadmin
+#gce_vm_service_account        = "apigee-admin-ws@anthos-bm-example1.iam.gserviceaccount.com"
+```
+
+4. Rename the `variables` file to default name used by Terraform for the `variables` file:
+> **Note:** You can skip this step if you run `terraform apply` with the `-var-file` flag
+```sh
+mv terraform.tfvars.sample terraform.tfvars
+```
+
+5. Navigate to the root directory of this repository initialize it as a Terraform directory
+```sh
+# this sets up the required Terraform state management configurations, similar to 'git init'
+terraform init
+```
+
+6. Create a _Terraform_ execution plan
+```sh
+# compares the state of the resources, verifies the scripts and creates an execution plan
+terraform plan
+```
+
+7. Apply the changes described in the _Terraform_ script
+```sh
+# executes the plan on the given provider (i.e: GCP) to reach the desired state of resources
+terraform apply
+```
+> **Note:** When prompted to confirm the Terraform plan, type 'Yes' and enter
+
+***The `apply` command sets up the Compute Engine VM based bare metal infrastructure. This can take a few minutes (approx. 3-5 mins) for the entire bare-metal cluster to be setup.***
+
+---
+### Deploy an Anthos cluster
+
+After the Terraform execution completes you are ready to deploy an Anthos cluster.
+
+1. SSH into the admin host
+```sh
+gcloud compute ssh tfadmin@apigee-cluster-abm-ws0-001 --project=<YOUR_PROJECT> --zone=<YOUR_ZONE>
+```
+
+2. Install the Anthos cluster on the provisioned Compute Engine VM based bare metal infrastructure
+```sh
+sudo ./run_initialization_checks.sh && \
+sudo bmctl create config -c apigee-cluster && \
+sudo cp ~/apigee-cluster.yaml bmctl-workspace/apigee-cluster && \
+sudo bmctl create cluster -c apigee-cluster && \
+```
+
+```
```

Comment:
Why split the command? Can we have it in the same shell section?

Reply:
Included in same command